GameOfRunes/GameOfRunes/GameEngine/SpawnDelegate.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  SpawnSystem.swift
+//  GameOfRunes
+//
+//  Created by Brian Yen on 28/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import GameplayKit
+
+/**
+ This class deals with the spawning of enemies. Level data is accessed via
+ `GameMetaData` and is updated at periodic time intervals.
+ */
+class SpawnDelegate {
+    private weak var gameEngine: GameEngine?
+    private weak var gameMetaData: GameMetaData?
+    private var timeTillNextSpawn: TimeInterval
```

Comment:
For this, we will want to reflect it to the screen also, so I think we will need to use a timer for it, and repeatedly countdown, so we should move it to game metadata, then in the future when we add a timer sknode, we can put it as part of player component and the player system will handle the synchronisation then you just fetch from metadata

Reply:
rmb to add a todo

Reply:
Okay will KIV this and keep as a TODO.